hs_core/models.py

```diff
@@ -1940,14 +1941,29 @@ def create(cls, resource, file, folder=None, source=None, move=False):
 
         else:  # if file is not an open file, then it's a basename (string)
             if file is None and source is not None:
-                root, file = os.path.split(source)  # take from source path
-            target = get_resource_file_path(resource, file, folder=folder)
-            if source is not None:
+                # source is a path to an iRODS file to be copied here.
+                root, newfile = os.path.split(source)  # take file from source path
+                # newfile is where it should be copied to.
+                target = get_resource_file_path(resource, newfile, folder=folder)
                 istorage = resource.get_irods_storage()
                 if not move:
                     istorage.copyFiles(source, target)
                 else:
                     istorage.moveFile(source, target)
+                if not istorage.exists(target):
```

Comment:
This is a check that the copy was done. It has to occur after the copy. 
The check before should be that source exists.